### ISSUE TITLE
[Application Request] AzuraCast

### ISSUE THREAD

**Issue by SlvrEagle23** [GhostWriters/DockSTARTer#1339]
# Name of the application

AzuraCast

## GitHub

https://github.com/AzuraCast/AzuraCast

## Docker Hub/GitHub Registry

https://hub.docker.com/u/azuracast
Alternate: https://github.com/orgs/AzuraCast/packages

### Additional context

Hello!

I'm the creator and lead developer of AzuraCast. AzuraCast is a self-hosted "web radio station in a box",  a free and open-source web application for creating and maintaining web radio stations combined with all of the necessary software all installed and configured to work together out of the box.

For years now, our primary installation method has been through Docker, specifically [a Docker Compose file](https://github.com/AzuraCast/AzuraCast/blob/main/docker-compose.sample.yml) that we keep updated on the root of our repository. We've built up a small little helper/utility script that does a lot of what DockSTARTer does, but in a much less elegant way. 

After discovering DockSTARTer, needless to say we see a ton of overlap in how we provision installations already and how this software works. I think we would be an _excellent_ candidate for inclusion in this software, and it would make our installation process so straightforward that even more people would be inclined to use our Docker-centric installation method.

Please let me know exactly what steps we'd need to take on our project's end to help facilitate our inclusion in the DockSTARTer ecosystem. We'd be more than happy to devote some resources to making any necessary changes to ensure things can be more smoothly integrated.

**Comment by nemchik** [comment]
At a quick glance the only concern I have is the compose example you linked above seems to multiple containers all setup at once, which is fine, but are they all required? We usually try to include the easiest option for a user to setup, even if it's not strictly done "the docker way" (one process per container, etc). So we often prefer an all in one container with the exception of things like a reverse proxy or a database.

I'm not saying this is a set in stone requirement, but trying to get the idea across to a user that they need redis and a number of other containers as dependents sometimes leads to additional support requests.

**Comment by SlvrEagle23** [comment]
@nemchik Fair enough, we've definitely entertained the idea of having everything run in basically one big container ourselves, as it would certainly be possible to do given how our application is setup.

Currently the nginx_proxy is of course not required (and can be substituted out for any other frontend proxy as desired), but the web container, Redis cache, MariaDB and the "stations" container (containing the actual broadcast software) are all required.

Would having the option of a singular "self-contained" container holding the whole application and all of its dependencies help with rolling it into DockSTARTer? If so, we can expedite our work on that.

Would any containers still be reasonable to have outside the main one, i.e. databases? (Redis for us is a performance-boosting cache for us that also stores app sessions, but nothing there needs to be persisted, so it could spin up and down with the service with no issue.)

**Comment by nemchik** [comment]
If everything but the database, reverse proxy, and redis were in a single container that should be easy enough. DS would need to add redis and your new image.

We have mariadb for the database, and swag for the reverse proxy.

**Comment by SlvrEagle23** [comment]
I've worked on some improvements to consolidating the docker containers used for AzuraCast, but unfortunately currently the best we can do for consolidation is to get it down to two containers: the web services one (running PHP/nginx/SFTP/etc) and the station services one (running Liquidsoap, Icecast, SHOUTcast, etc).

We did add support for operating without Redis as a necessary dependency (by passing `ENABLE_REDIS=false` as an env var to the web container), so at this point only 3 containers should be needed:

 - `ghcr.io/azuracast/web:latest` (or `:stable` for stable releases)
 - `ghcr.io/azuracast/radio:latest` (or `:stable` for stable releases)
 - a MariaDB container (possibly managed centrally for multiple apps)

The `web` container expects to see the `radio` container with the name `stations` (via a link, for example), and the MariaDB connection info is fully configurable via environment vars as well.

We're continuing to work on optimizing this, but hopefully this will be sufficient for us to be able to be included in DockSTARTer's environment, as it would be an excellent way for users to get started with our software.

**Comment by angrycuban13** [comment]
@SlvrEagle23 we will look into adding it but PRs are welcome too!

**Comment by SlvrEagle23** [comment]
@angrycuban13 My apologies, bit of a silly moment on my part...I couldn't find the various app manifests in this repo and so I assumed they were managed somewhere else, but I just wasn't looking in the `.apps` folder since I thought it was internal/hidden.

I'll get a PR in shortly. :)

**Comment by SlvrEagle23** [comment]
Update: I'm taking a look at things, and admittedly I'm having trouble finding any example where DockSTARTer is spinning up a single project with more than one container/image specified. I can gather how it's assembling things, but I'm wondering if it would actually pose any sort of problem for a single service to actually run on multiple containers.

**Comment by nemchik** [comment]
Sorry for not getting back to you about this sooner. I haven't had much time to work on this project lately.

> Update: I'm taking a look at things, and admittedly I'm having trouble finding any example where DockSTARTer is spinning up a single project with more than one container/image specified. I can gather how it's assembling things, but I'm wondering if it would actually pose any sort of problem for a single service to actually run on multiple containers.

Thus far we've avoided adding any such apps to DS, since the initial goal of the project was to help people get off the ground and with running applications and to be easy enough to understand how things are run for those inclined to look beyond the GUI. I will have to look around at the code and see if we truly have anything that has a hard requirement of not having two services in one single app template folder, but it would probably look pretty unique.

> The `web` container expects to see the `radio` container with the name `stations` (via a link, for example), and the MariaDB connection info is fully configurable via environment vars as well.

The naming of the two containers (that allows them to communicate to each other) might be something we would need to be configurable. I would likely want to name the containers `azuracast_web` and `azuracast_radio` or `azuracast_station`. Basically we want the container name to reflect the name of the service it's running in some way so the user and DS itself can recognize it and manage it accordingly.

**Comment by SlvrEagle23** [comment]
@nemchik That shouldn't be a problem...I believe the Compose specification lets you specify link aliases between containers, so no matter what it's named it'll be seen by one container as the correct name. That's even how we have things configured on our stock installation, since our containers are also named things like `azuracast_web` and `azuracast_stations`.
